### PR TITLE
SPMD ForList support

### PR DESCRIPTION
**Pull request by zhangwen0411**

### CODE REVIEW

File: plugin/src/regent/flow_extract_subgraph.t [hunk 1]
```diff
@@ -27,7 +27,8 @@
 -- (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
 -- OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
 
--- Extract Node into a Subgraph
+-- Extract Node into a Subgraph containing the Node itself, all adjacent
```
Comment (zhangwen0411) [inline plugin/src/regent/flow_extract_subgraph.t]: adjacent data nodes

File: plugin/src/regent/flow_extract_subgraph.t [hunk 2]
```diff
@@ -48,6 +49,8 @@ function context.new_global_scope()
   return setmetatable(cx, context)
 end
 
+-- Extracts node `nid` into a subgraph.
+-- Returns the subgraph and the node's nid in the subgraph.
```
Comment (zhangwen0411) [inline plugin/src/regent/flow_extract_subgraph.t]: new nid

File: plugin/src/regent/flow_spmd.t [hunk 3]
```diff
@@ -2702,12 +2752,11 @@ local function rewrite_shard_intersections(cx, shard_loop, intersections)
   return shallow_intersections, mapping
 end
 
-local function rewrite_inner_loop_bounds(cx, loop_nid, start)
+local function rewrite_inner_fornum_loop_bounds(cx, loop_nid, shard_part_indices)
```
Comment (zhangwen0411) [inline plugin/src/regent/flow_spmd.t]: Factor this out?

File: plugin/src/regent/flow_spmd.t [hunk 4]
```diff
@@ -4484,36 +4940,23 @@ local function issue_barrier_creation(cx, rhs_nid, intersection_nid,
...
+  local nparts_label = cx.graph:node_label(nparts_nid)
   local create_label = flow.node.Opaque {
     action = ast.typed.stat.Var {
       symbols = terralib.newlist({collective.value.value}),
       types = terralib.newlist({collective_type}),
       values = terralib.newlist({
           ast.typed.expr.DynamicCollective {
             arrivals = ast.typed.expr.Binary {
-              lhs = ast.typed.expr.Binary {
-                lhs = ast.typed.expr.Binary {
```
Comment (zhangwen0411) [inline plugin/src/regent/flow_spmd.t:4503]: This should be `nparts`.

**Comment by elliottslaughter** [comment]
Merged.

**Pull request closed without merging**